DTXMania/Code/Score,Song/CBoxDef.cs

```diff
@@ -161,6 +158,34 @@ public CBoxDef( string boxdefファイル名 )
 									this.Difficlty = b;
 								}
 							}
+							else
+							{
+								// hit ranges
+								// map the legacy hit ranges to apply to each category
+								// they should only appear when reading from a legacy box.def,
+								// so simply copy values over whenever there is a change
+								STHitRanges stLegacyHitRanges = new STHitRanges(nDefaultSizeMs: -1);
+								if (tTryReadHitRangesField(str, string.Empty, ref stLegacyHitRanges))
+								{
+									stDrumHitRanges = STHitRanges.tCompose(stLegacyHitRanges, stDrumHitRanges);
+									stDrumPedalHitRanges = STHitRanges.tCompose(stLegacyHitRanges, stDrumPedalHitRanges);
+									stGuitarHitRanges = STHitRanges.tCompose(stLegacyHitRanges, stGuitarHitRanges);
+									stBassHitRanges = STHitRanges.tCompose(stLegacyHitRanges, stBassHitRanges);
+									continue;
+								}
+
+								if (tTryReadHitRangesField(str, @"DRUM", ref stDrumHitRanges))
+									continue;
+
+								if (tTryReadHitRangesField(str, @"DRUMPEDAL", ref stDrumPedalHitRanges))
+									continue;
+
+								if (tTryReadHitRangesField(str, @"GUITAR", ref stGuitarHitRanges))
+									continue;
+
+								if (tTryReadHitRangesField(str, @"BASS", ref stBassHitRanges))
+									continue;
```

Comment:
We are moving away from the standard box.def file format which is shared by all DTXMania derivatives, documented here: https://osdn.net/projects/dtxmania/wiki/box.def%20format
I think it's good, but it would be good to double check with @DTXMania (FROM) and update the official documentation, saying it is a DTXManiNX specific feature.
In the meantime, you can document the improved format here: https://github.com/DTXManiaNX/Documentation/wiki

Reply:
Added an initial version of the documentation [here](https://github.com/DTXManiaNX/Documentation/wiki/Music-boxes).